shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -16,23 +16,66 @@
 
 package com.nvidia.spark.rapids.shims.spark310
 
+import java.io.ByteArrayInputStream
+import java.nio.{ByteBuffer, ByteOrder}
+
 import scala.collection.JavaConverters._
 import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
 
 import ai.rapids.cudf._
 import com.nvidia.spark.rapids._
 import com.nvidia.spark.rapids.GpuColumnVector.GpuColumnarBatchBuilder
 import com.nvidia.spark.rapids.RapidsPluginImplicits._
+import org.apache.avro.util.ByteBufferInputStream
+import org.apache.hadoop.conf.Configuration
+import org.apache.parquet.HadoopReadOptions
+import org.apache.parquet.hadoop.{ParquetFileReader, ParquetInputFormat}
+import org.apache.parquet.io.{DelegatingSeekableInputStream, InputFile, SeekableInputStream}
 
 import org.apache.spark.TaskContext
 import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
+import org.apache.spark.sql.catalyst.expressions.codegen.GenerateUnsafeProjection
+import org.apache.spark.sql.catalyst.util.DateTimeUtils
 import org.apache.spark.sql.columnar.{CachedBatch, CachedBatchSerializer}
+import org.apache.spark.sql.execution.datasources.DataSourceUtils
+import org.apache.spark.sql.execution.datasources.parquet.{ParquetReadSupport, ParquetRecordMaterializer, ParquetToSparkSchemaConverter, ParquetWriteSupport}
 import org.apache.spark.sql.internal.SQLConf
-import org.apache.spark.sql.types.{StructField, StructType}
+import org.apache.spark.sql.internal.SQLConf.LegacyBehaviorPolicy
+import org.apache.spark.sql.types._
 import org.apache.spark.sql.vectorized.ColumnarBatch
 import org.apache.spark.storage.StorageLevel
+import org.apache.spark.util.SerializableConfiguration
+
+class ByteArrayInputFile(buff: Array[Byte]) extends InputFile {
+
+  private class SeekableByteArrayInputStream(buff: Array[Byte]) extends ByteArrayInputStream(buff) {
+    def getSeekableStream: SeekableInputStream = {
+      // ParquetFileReader reads in little endian! :shrugs
+      val byteBuffer = ByteBuffer.wrap(buff).order(ByteOrder.LITTLE_ENDIAN)
```

Comment:
Why are we changing the byte order only when getting a seekable stream?  If the ordering does need to be changed, why not specify it up-front when wrapping the buffer with a `ByteArrayInputStream`?  Otherwise someone could call the inherited `ByteArrayInputStream` methods on the object and get different behavior.